### PR TITLE
Add Partis.si torrent tracker

### PR DESCRIPTION
**Pull request by Lenart12**
#### Indexer/Tracker

Partis.si

#### Description
Add a new semi-private slovenian torrent tracker

### CODE REVIEW

File: definitions/v7/partis.yml [hunk 1]
```diff
@@ -0,0 +1,189 @@
...
+
+search:
+  paths:
+    - path: /torrent/search
+  inputs:
+    q: "{{ .Keywords }}"
+    cat: "{{join .Categories \",\"}}"
+    order: "{{ .Config.sort }}"
+
+  preprocessingfilters:
```
Comment (bakerboy448) [inline definitions/v7/partis.yml:144]: Html site that is scraping but returns json in the html?

Or can you use the existing Cardigann json support?

**Review by bakerboy448 — CHANGES_REQUESTED** [review]
various lint and schema issues to fix

File: definitions/v7/partis.yml (hunk 1 above)
Comment (Lenart12) [reply definitions/v7/partis.yml:144]: Actual response is not "json" it's more like javascript - i.e. the actual response is
```js
data = {{"type":"text/plain; charset=utf-8","config":{"tpl":"browse"},"data":{},"values":{},"elements":{},"sets":{"torrent_list":{"cols":["fid","w","created_at","title","seeders","leechers","short_description","torrent_category_id","completed","size","icon","image"],"data":[
// The scraped data
], ... more junk}

window.onscroll = torrentScroll;

function arJs(){

}
```
Can't use json support as there is no way to clean up the response - remove text before and after the json array.

**Comment by Lenart12** [comment]
Fixed lint and schema

File: definitions/v7/partis.yml [hunk 2]
```diff
@@ -0,0 +1,192 @@
+---
+id: partis
+name: Partis
+description: >
```
Comment (bakerboy448) [inline definitions/v7/partis.yml:4]: Wrap in `"` and make a single line

File: definitions/v7/partis.yml [hunk 3]
```diff
@@ -0,0 +1,192 @@
...
+      selector: short_description
+    date:
+      selector: created_at
+    size:
+      selector: size
+    leechers:
+      selector: leechers
+    seeders:
+      selector: seeders
+
```
Comment (bakerboy448) [inline definitions/v7/partis.yml:185]: Nit: uneeded new line

File: definitions/v7/partis.yml [hunk 4]
```diff
@@ -0,0 +1,192 @@
...
+      'seeders ASC': 'Seeders ascending'
+      'seeders DESC': 'Seeders descending'
+      'leechers ASC': 'Leechers ascending'
+      'leechers DESC': 'Leechers descending'
+      'size ASC': 'Size ascending'
+      'size DESC': 'Size descending'
+
+login:
+  method: post
+
```
Comment (bakerboy448) [inline definitions/v7/partis.yml:124]: Nit: no new line

File: definitions/v7/partis.yml [hunk 5]
```diff
@@ -0,0 +1,192 @@
...
+  path: /user/login
+
```
Comment (bakerboy448) [inline definitions/v7/partis.yml:126]: same

File: definitions/v7/partis.yml [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    search: [q]
+    tv-search: [q]
+    movie-search: [q]
+    music-search: [q]
+    book-search: [q]
+
+settings:
+  - name: username
+    type: text
+    label: Username
```
Comment (bakerboy448) [inline definitions/v7/partis.yml:92]: nit: no new line

File: definitions/v7/partis.yml [hunk 7]
```diff
@@ -0,0 +1,192 @@
...
+
+  - name: password
+    type: password
+    label: Password
+
```
Comment (bakerboy448) [inline definitions/v7/partis.yml:97]: same

File: definitions/v7/partis.yml [hunk 8]
```diff
@@ -0,0 +1,192 @@
...
+  # Broken on website for now
+  # - name: freeleech
+  #   type: checkbox
+  #   label: Search freeleech only
+  #   default: false
+
```
Comment (bakerboy448) [inline definitions/v7/partis.yml:103]: Same

File: definitions/v7/partis.yml [hunk 9]
```diff
@@ -0,0 +1,192 @@
...
+  paths:
+    - path: /torrent/search
+  inputs:
+    q: "{{ .Keywords }}"
+    cat: "{{join .Categories \",\"}}"
+    order: "{{ .Config.sort }}"
+
+  preprocessingfilters:
+    # Extract torrent information from the json array that is part
+    # of the response and save transform it into html form
```
Comment (bakerboy448) [inline definitions/v7/partis.yml:148]: no new line; one line

**Review by bakerboy448 — CHANGES_REQUESTED** [review]
couple fixes otherwise looks good

I assume tested?

**Comment by Lenart12** [comment]
Fixed more nits :^)

Working, also tested torrent download:
image
image

**Review by bakerboy448 — APPROVED** [review]

**Pull request merged**